### PR TITLE
8273101: Eliminate the usage of threadgroup sandboxing in the java.util.logging

### PR DESCRIPTION
**Pull request by mrserb**
At the time Java supported applets and webstart, a special mechanism for launching various applications in one JVM was used to reduce memory usage and each application was isolated from each other.

This isolation was implemented via ThreadGroups where each application created its own thread group and all data for the application was stored in the context of its own group.

Some parts of the JDK used ThreadGroups directly, others used special classes like sun.awt.AppContext.

Such sandboxing became useless since the plugin and webstart are no longer supported and were removed in jdk11.

Note that this is just a first step for the overall cleanup, here is my plan:
1. Cleanup the usage of AppContext in the “java.util.logging.LogManager" class in the java.base module.
2. Cleanup the usage of TheadGroup in the JavaSound
3. Cleanup the usage of TheadGroup in the JavaBeans
4. Cleanup the usage of AppContext in the Swing
5. Cleanup the usage of AppContext in the AWT
6. Delete the AppContext class.

The current PR is for the first step. So this is the request to delete the usage of AppContext in the LogManager and related tests.

Tested by tier1/tier2/tier3 and jdk_desktop tests.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8273101](https://bugs.openjdk.java.net/browse/JDK-8273101): Eliminate the usage of threadgroup sandboxing in the java.util.logging

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5326/head:pull/5326` \
`$ git checkout pull/5326`

Update a local copy of the PR: \
`$ git checkout pull/5326` \
`$ git pull https://git.openjdk.java.net/jdk pull/5326/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5326`

View PR using the GUI difftool: \
`$ git pr show -t 5326`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5326.diff">https://git.openjdk.java.net/jdk/pull/5326.diff</a>

</details>

### CODE REVIEW

**Comment by prrace** [comment]
This fix requires coordinated closed changes so needs an Oracle sponsor.
And actually is probably better handled entirely by an Oracle engineer because pushes need to be very co-ordinated.

**Comment by prrace** [comment]
I believe we should have a CSR for this. It removes a long standing capability in the platform that was used by
components such as plugin and webstart.

/csr

**Comment by prrace** [comment]
Hmm I was under the impression this was removing AppContext itself but it is just removing the backdoor needed by logging
Perhaps this isn't the change that requires the CSR but it then leaves an inconsistent state where desktop supports AppContext still but other modules don't ...

**Comment by prrace** [comment]
@aivanov-jdk will help make sure the closed changes are pushed at exactly the same time as this gets pushed.

**Comment by mrserb** [comment]
> Perhaps this isn't the change that requires the CSR but it then leaves an inconsistent state where desktop supports AppContext still but other modules don't ...

Even java.desktop does not support it fully, since for a couple of years nobody care about appcontext when write a new code.
Note that I mentioned the "threadgroup sandboxing" in the subject, which is not necessary implemented via Appcontext. The JavaBeans and JavaSound use the thread group directly, I plan to remove them as well.

**Comment by dfuch** [comment]
Right. I am also a bit uncomfortable about the inconsistency. That said - if everybody agrees that this should be done I have no objection. The changes to the java.util.logging implementation correspond to my expectation if this were to be carried through.

File: test/jdk/java/util/logging/TestLoggingWithMainAppContext.java [hunk 1]
```diff
@@ -1,77 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (mrserb) [reply test/jdk/java/util/logging/TestLoggingWithMainAppContext.java:1]: It uses it indirectly, the next line under security manager trigger creation of the appcontext:
`ImageIO.read(is); // triggers calls to system loggers & creation of main AppContext`

but I can preserve/rename it for sure.

**Comment by AlanBateman** [comment]
No objection to removing this legacy/unused code but I think it would be useful to useful to have the JBS issue or the PR summary provide a bit more context. As I see it, this is just one piece of the overall cleanup and I assume there are more substantive changes to the java.desktop module coming, is that right?

**Comment by mrserb** [comment]
> As I see it, this is just one piece of the overall cleanup and I assume there are more substantive changes to the java.desktop module coming, is that right?

Yes, you are right.

**Comment by mlchung** [comment]
> Hmm I was under the impression this was removing AppContext itself but it is just removing the backdoor needed by logging
> Perhaps this isn't the change that requires the CSR but it then leaves an inconsistent state where desktop supports AppContext still but other modules don't ...

What is the plan to remove AppContext completely from java.desktop?   It's okay to have a separate PR to remove the logging dependency on AppContext but I'd prefer to see AppContext be removed about the same time with this PR.

**Comment by mrserb** [comment]
I'll remove its used one by one from the external usage like in "java.base" here, then in Swing, then in 2D like fonts, then last in AWT.

That change should not break something other than the tests which intentionally create different appcontexts. And it's easy to review.

**Comment by mrserb** [comment]
> No objection to removing this legacy/unused code but I think it would be useful to useful to have the JBS issue or the PR summary provide a bit more context. As I see it, this is just one piece of the overall cleanup and I assume there are more substantive changes to the java.desktop module coming, is that right?

I have updated the PR and JBS

**Comment by mrserb** [comment]
Does anybody have any other suggestions?

**Comment by mrserb** [comment]
> The change looks okay. My suggestion is to get 1-6 all ready to push around the same time. It's okay to have separate JBS issues and PRs.

ok, I'll continue to work using the plan from the description.

**Pull request closed without merging**